libs/island-ui/core/src/lib/DatePicker/DatePicker.tsx

```diff
@@ -0,0 +1,142 @@
+import React, { useEffect, useState } from 'react'
+import cn from 'classnames'
+import {
+  default as ReactDatePicker,
+  registerLocale,
+  ReactDatePickerProps,
+} from 'react-datepicker'
+import { format, getYear } from 'date-fns'
+import pl from 'date-fns/locale/pl'
+import is from 'date-fns/locale/is'
+import en from 'date-fns/locale/en-US'
+
+import Icon from '../Icon/Icon'
+import Typography from '../Typography/Typography'
+
+import * as styles from './DatePicker.treat'
+import 'react-datepicker/dist/react-datepicker.css'
+import './overwriteStyles.css'
+
+type Locale = 'is' | 'pl'
+interface DatePickerProps {
+  label: string
+  placeholderText: ReactDatePickerProps['placeholderText']
+  locale?: Locale
+  value?: ReactDatePickerProps['value']
+  minDate?: ReactDatePickerProps['minDate']
+  handleChange?: () => void
+  onInputClick?: ReactDatePickerProps['onInputClick']
+}
+
+export const DatePicker: React.FC<DatePickerProps> = ({
+  label,
+  placeholderText,
+  locale,
+  value,
+  minDate,
+  handleChange,
+  onInputClick,
+}) => {
+  const [startDate, setStartDate] = useState<Date | null>(null)
+  const [datePickerState, setDatePickerState] = useState<'open' | 'closed'>(
+    'closed',
+  )
+  const className = cn(
+    styles.inputContainer,
+    styles.inputContainerVariants[datePickerState],
+  )
+
+  useEffect(() => {
+    if (locale === 'is') {
+      registerLocale('is', is)
+    } else if (locale === 'pl') {
+      registerLocale('pl', pl)
+    }
+  }, [locale])
+
+  const getLocale = (locale: Locale) => {
+    return locale === 'is' ? is : locale === 'pl' ? pl : en
+  }
+
+  const CustomInput = ({ value, onClick, placeholderText }) => {
+    const valueAsDate = new Date(value)
+
+    return (
+      <button className={className} onClick={onClick}>
```

Comment:
any reason why you are not using the components from island-ui here?

Reply:
The thinking is that it would require an extra div to be able to set a classname.

Reply:
Yet another reason for all core components to be able to receive a className.

Reply:
@DavidBachmann Do you want core components to receive className? That's really scary for me since devs could use that prop to stretch the components in dozens of ways. Very difficult to maintain going forward.

There are imo better ways to handle this, depending on each case.

* Wrap with div.
* Add more variants to core component.
* Duplicate component, with or without shared treat styles.